storage/store.go

```diff
@@ -647,12 +647,14 @@ func (s *Store) MaybeGossipConfigs() error {
 		if _, err := rng.getLeaseForGossip(s.Context(nil)); err != nil {
 			return err
 		}
-		// Always gossip the configs as there are leader lease acquisition
-		// scenarios in which the config is not gossiped (e.g. leader executes
-		// but crashes before gossiping; comes back up but group goes dormant).
-		rng.maybeGossipConfigs(func(configPrefix proto.Key) bool {
-			return bytes.Equal(configPrefix, cd.keyPrefix)
-		})
+		// Configs are gossiped immediately when forced. Otherwise, configs
+		// will be gossiped when they are updated or a leader lease is
+		// acquired/extended.
+		if force {
+			rng.maybeGossipConfigs(func(configPrefix proto.Key) bool {
+				return bytes.Equal(configPrefix, cd.keyPrefix)
+			})
+		}
```

Comment:
When there's no lease, the store attempts to acquire the lease immediately. This change is for the case where a lease has been granted but the lease holder has restarted. It's mainly a test issue, although it could become relevant in production if our lease timeouts were longer. I think it's a fairly clean fix: any time we want "do this when you have the lease", we need to do it both in InternalLeaseRequest when the lease is granted, and at startup if loadLeaderLease tells us we already have the lease.